sdk/go/common/util/gitutil/git.go

```diff
@@ -255,9 +301,101 @@ func parseAuthURL(remoteURL string) (string, *http.BasicAuth, error) {
 	return u.String(), auth, nil
 }
 
+// sshConfig allows tests to mock SSH key paths.
+type sshConfig struct {
+	path string
+	err  error
+}
+
+// GetKeyPath reads SSH configuration from disk when the receiver is nil. When
+// non-nil it acts as a mock.
+func (c *sshConfig) GetKeyPath(host string) (string, error) {
+	if c == nil {
+		return ssh_config.GetStrict(host, "IdentityFile")
+	}
+	return c.path, c.err
+}
```

Comment:
This seems like a hand-rolled mock for a top-level function.
ssh_config.GetStrict just calls [DefaultUserSettings](https://pkg.go.dev/github.com/kevinburke/ssh_config#DefaultUserSettings).GetStrict.

Can we instead just declare an interface for UserSettings?

```go
type sshUserSettings interface {
  GetStrict(alias, key string) (string, error)
}

var _ sshUserSettings = (*ssh_config.UserSettings)(nil)
```

And in `getSSHPublicKeys`:

```go
func getSSHPublicKeys(..., sshConfig sshUserSettings) (...) {
  if sshConfig == nil { sshConfig = ssh_config.DefaultUserSettings }

  // ..
}
```

That leaks less of the test code into non-test.

Reply:
I did this and moved the struct into the test file (renamed to `mockSSHConfig`) -- let me know if you had something else in mind!